### DISCUSSION TITLE
Essentials commands in command blocks (Handling selectors)

### DISCUSSION THREAD

**Discussion by Roos-Skywalker** [EssentialsX/Essentials#4916] [Ideas and Feature Suggestions] [answered]
It would be very nice to be able to use Essentials commands in command blocks again. Apparently this used to be a feature, but it is gone now.

Having this feature would allow server owners to do more with less plugins. Not only that, it also affects existing commands such as /tell.
Despite being a vanilla command, because Essentials also has its own whisper system, the command blocks cannot whisper to players.
No, using /msg (not used by EssentialsX but does the same thing) instead /tell did not work.

Selectors I have tried that resulted in no player being found:
`@a, @p, @a[distance=..2], /execute as @a[distance..2] run warp tutorial @s`

Selectors that did work but affected all players:
`*, **`

**Comment by pop4959** [comment, marked as answer]
Selectors have never been a feature native to Essentials. It used to be possible on older versions, but only because the server allowed it.

It's unlikely that we will be adding explicit support for selectors ourselves in the foreseeable future, but you can download plugins like [CommandHook](https://www.spigotmc.org/resources/commandhook.61415/) which restore the behavior of allowing selectors in plugins.

Note that `*` and `**` are not vanilla selectors, these have always existed in Essentials.

You can also run vanilla Minecraft commands even with Essentials installed, by prepending `minecraft:` to the command. For example, `/minecraft:tell` instead of `/essentials:tell` or just `/tell` (which defaults to plugin commands over vanilla).

**Reply by Roos-Skywalker** [reply]
Thank you very much! I found a different solution for my problem but I really appreciate having multiple options to solve it. For future Google searches here's my solution: It specifically reverts command block commands to vanilla with the commands you specify. In this case /tell.

1. Open commands.yml (inside your server's root folder) with a proper text editor such as [Notepad++](https://notepad-plus-plus.org/downloads/v8.4.1/).
2. Enter the command you want vanilla behaviour for under `command-block-overrides:`. For example to make command blocks use the vanilla /tell command it should look like this:
 `command-block-overrides:`
`- tell`

If it looks like `command-block-overrides[]` instead, just overwrite it with the example code snippet above.
